File: solutions/r/sum-of-multiples/sum-of-multiples.R
# Adds up all numbers from 1 to a given end number that are multiples
# of the factors provided.
sum_of_multiples <- function(factors, limit) {
    numbers <- c(0)
    for (factor in factors[factors <= limit]) {
        numbers <- c(numbers, seq(factor, limit - 1, factor))
    }
    sum(unique(numbers))
}

File: solutions/elixir/sum-of-multiples/lib/sum_of_multiples.ex
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples
  of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, factors) do
    factors
    |> Enum.reject(&(&1 == 0))
    |> Enum.map(&Range.new(&1, limit - 1, &1))
    |> Enum.map(&Enum.to_list/1)
    |> List.flatten()
    |> Enum.uniq()
    |> Enum.sum()
  end
end
